app/src/main/java/movierating/services/ReviewServiceImpl1.java

```diff
@@ -15,7 +18,21 @@ public ReviewServiceImpl1(UserRepository userRepository, MovieRepository movieRe
 
 
     @Override
-    public void addReview() {
+    public void addReview(String userName, String movieName, int rating) {
+
+        User user = this.userRepository.getUserList().get(userName);
```

Comment:
'this.userRepository.getUserList()' here can give null as well. In our case it does not, but since userRepository is public. You should not assume it to not give null. You should be handling the case where 'this.userRepository.getUserList()'  is null. Other wise it can give Null pointer exceptions.

Reply:
Got it. Added.